include/pasta/AST/Stmt.h

```diff
@@ -1136,13 +1136,13 @@ class CXXForRangeStmt : public Stmt {
   PASTA_DECLARE_BASE_OPERATORS(Stmt, CXXForRangeStmt)
   std::vector<::pasta::Stmt> Children(void) const;
   ::pasta::Token BeginToken(void) const;
-  ::pasta::DeclStmt BeginStatement(void) const;
+  std::optional<::pasta::DeclStmt> BeginStatement(void) const;
```

Comment:
What are `CXXCatchStmt::BeginStatement` and `EndStatement`? In Clang I imagine we can use these for iteration, but probably not here?

Reply:
In clang we can use it to get `Begin` and `End` stmt during iterations and here PASTA apis provide the wrapped begin and end stmt. It gets used during serialization of `CXXForRangeStmt`.